lib/expression.ts

```diff
@@ -11,31 +17,71 @@ export default class ExpressionTranspiler extends base.TranspilerBase {
       case ts.SyntaxKind.BinaryExpression:
         var binExpr = <ts.BinaryExpression>node;
         var operatorKind = binExpr.operatorToken.kind;
-        if (operatorKind === ts.SyntaxKind.EqualsEqualsEqualsToken ||
-            operatorKind === ts.SyntaxKind.ExclamationEqualsEqualsToken) {
-          if (operatorKind === ts.SyntaxKind.ExclamationEqualsEqualsToken) this.emit('!');
-          this.emit('identical (');
-          this.visit(binExpr.left);
-          this.emit(',');
-          this.visit(binExpr.right);
-          this.emit(')');
-        } else {
-          this.visit(binExpr.left);
-          if (operatorKind === ts.SyntaxKind.InstanceOfKeyword) {
+        var tokenStr = ts.tokenToString(operatorKind);
+        switch (operatorKind) {
+          case ts.SyntaxKind.EqualsEqualsEqualsToken:
+          case ts.SyntaxKind.ExclamationEqualsEqualsToken:
+            if (operatorKind === ts.SyntaxKind.ExclamationEqualsEqualsToken) this.emit('!');
+            this.emit('identical (');
+            this.visit(binExpr.left);
+            this.emit(',');
+            this.visit(binExpr.right);
+            this.emit(')');
+            break;
+          case ts.SyntaxKind.CaretToken:
+          case ts.SyntaxKind.BarToken:
+          case ts.SyntaxKind.AmpersandToken:
+          case ts.SyntaxKind.GreaterThanGreaterThanToken:
+          case ts.SyntaxKind.LessThanLessThanToken:
+          case ts.SyntaxKind.CaretEqualsToken:
+          case ts.SyntaxKind.BarEqualsToken:
+          case ts.SyntaxKind.AmpersandEqualsToken:
+          case ts.SyntaxKind.GreaterThanGreaterThanEqualsToken:
+          case ts.SyntaxKind.LessThanLessThanEqualsToken:
+            // we special case this for Dart since dart2ts will
+            // convert `number` to `num` and if there are any numbers
+            // that are evaluated with bitwise operations then the `num`
+            // number type will be handled properly. The code below will
+            // explicitly cast the associated values into ints temporarily.
+
+            if (tokenStr[tokenStr.length - 1] == "=") {
+              // assignment case (ident [op]=)
```

Comment:
add a comment on what this is doing? `For assignments, strip the trailing = so that...`